src/nextjournal/clerk/builder.clj

```diff
@@ -139,18 +139,26 @@
         _ (report-fn {:stage :init :state state})
         {state :result duration :time-ms} (eval/time-ms (mapv (comp (partial parser/parse-file {:doc? true}) :file) state))
         _ (report-fn {:stage :parsed :state state :duration duration})
-        {state :result duration :time-ms} (eval/time-ms (mapv (comp analyzer/hash
-                                                                    analyzer/build-graph) state))
+        {state :result duration :time-ms} (eval/time-ms
+                                           (try
+                                             (mapv (fn [{:as doc :keys [file]}]
+                                                     (try
+                                                       (analyzer/hash (analyzer/build-graph doc))
+                                                       (catch Exception e
+                                                         (throw (ex-info (str "cannot analyze notebook: " file) {:file file} e))))) state)))
         _ (report-fn {:stage :analyzed :state state :duration duration})
         _ (when download-cache-fn
             (report-fn {:stage :downloading-cache})
             (let [{duration :time-ms} (eval/time-ms (download-cache-fn state))]
               (report-fn {:stage :done :duration duration})))
-        state (mapv (fn [doc]
+        state (mapv (fn [{:as doc :keys [file]}]
                       (report-fn {:stage :building :doc doc})
                       (let [{doc+viewer :result duration :time-ms} (eval/time-ms
-                                                                    (let [doc (eval/eval-analyzed-doc doc)]
-                                                                      (assoc doc :viewer (view/doc->viewer (assoc opts :inline-results? true) doc))))]
+                                                                    (try
+                                                                      (let [doc (eval/eval-analyzed-doc doc)]
+                                                                        (assoc doc :viewer (view/doc->viewer (assoc opts :inline-results? true) doc)))
+                                                                      (catch Exception e
+                                                                        (throw (ex-info (str "cannot build notebook: " file) {:file file} e)))))]
```

Comment:
This might be not strictly necessary, given the printing of the notebook path via `report-fn`, but might be more easily discoverable and closer to the exception. It also adds the file as ex-data.